### ISSUE TITLE
Dotted variable in inferred type

### ISSUE THREAD

**Issue by nad** [agda/agda#2776]
Consider the following code:
```agda
data C : Set₁ where
  c : Set → C

data D : Set where
  d : D

F : C → Set₁
F (c _) = G d
  module M where
  G : D → Set₁
  G _ = Set
```
If the type of `M.G` is inferred, then Agda prints `(.x : Set) → D → Set₁`, which is not a valid expression.

This is a regression: Agda 2.5.1.1 prints `(_ : Set) → D → Set₁`, while Agda 2.5.2 produces the string mentioned above.

**Comment by nad** [comment]
Bisection suggests that the problem was introduced by @andreasabel in 637d485489327f398ca38c2ebbad8ac846e45374 (a bug fix for #1848):
```
$ cat script
#!/bin/sh

AGDA=$1

$AGDA --interaction --ignore-interfaces <<EOF
IOTCM "Bug.agda" None Indirect (Cmd_load "Bug.agda" [])
IOTCM "Bug.agda" None Indirect (Cmd_infer_toplevel Simplified "M.G")
EOF
$ agda-bisect --log log --good v2.5.1.1 --bad v2.5.2 --must-not-output "(.x : Set)" --compiler <ghc-7.10.3> --script ./script
[...]
637d485489327f398ca38c2ebbad8ac846e45374 is the first bad commit
commit 637d485489327f398ca38c2ebbad8ac846e45374
Author: Andreas Abel <andreas.abel@ifi.lmu.de>
Date:   Tue May 10 23:46:42 2016 +0200

    Fixed #1848 by naming unnamed pattern variables "x".

[...]
```

**Comment by nad** [comment]
My guess is that the commit above is not to blame, and that the problem is related to conversion from internal to concrete syntax.

**Comment by andreasabel** [comment]
Probably this is best fixed in `AbstractToConcrete`, like #2729.  
In this specific case, the inferred type could be printed as `Set -> D -> Set1`; it is not a user-written `Pi`-type, hence, it could be converted into a `Fun`.

**Comment by UlfNorell** [comment]
The problem exists also in 2.5.1.1:

```agda
data C : Set₁ where
  c : Set → C

data D : Set where
  d : D

F : C → {d : D} → C → Set₁
F x (c _) = G d
  module M where
  G : D → Set₁
  G _ = Set

-- With 2.5.1.1:
-- M.G : (x : C) {.d : D} (_ : Set) → D → Set₁
```
